pubpub/platform
# Ugrade to next-14 and next-runtime-env 3, making using NEXT_PUBLIC_ urls easier

- chore: update to next 14
- fix: use new next-runtim-env system
- chore: remove experimental serverActions setting, it's included by default now
- chore: update ts-rest to match the latest next version
- chore: replace sinful use of process.env

## Issue(s) Resolved

## Test Plan

At the very least do a test like this

1. Make sure you have nothing in `.env.local`, or `.env.production` (can be commented out.
2. run `pnpm build && pnpm start` (in core)
3. Witness app failing, especially noting that the `NEXT_PUBLIC` env vars are not present
4. Uncomment all variables in `.env.local`
5. `pnpm start`
6. Go to app
7. Witness it working

This proves that the `NEXT_PUBLIC` env vars are being read dynamically, which is what we want.

## Screenshots (if applicable)

## Optional

### Notes/Context/Gotchas

### Supporting Docs
